dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/trigramsummary/TrigramSummaryController.java

```diff
@@ -0,0 +1,133 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.trigramsummary;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import lombok.AllArgsConstructor;
+
+import org.hisp.dhis.common.DhisApiVersion;
+import org.hisp.dhis.fieldfiltering.FieldFilterService;
+import org.hisp.dhis.node.Preset;
+import org.hisp.dhis.security.acl.AclService;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.trackedentity.TrackedEntityAttributeService;
+import org.hisp.dhis.trackedentityattributevalue.TrackedEntityAttributeTableManager;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.hisp.dhis.webapi.service.ContextService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/**
+ * Trigram Summary endpoint to get a summary of all the trigram indexes and
+ * indexable attributes
+ *
+ * @author Ameen Mohamed
+ */
+@Controller
+@RequestMapping( value = TrigramSummaryController.RESOURCE_PATH )
+@ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
+@AllArgsConstructor
+public class TrigramSummaryController
+{
+    public static final String RESOURCE_PATH = "/trigramSummary";
+
+    private final TrackedEntityAttributeService trackedEntityAttributeService;
+
+    private final TrackedEntityAttributeTableManager trackedEntityAttributeTableManager;
+
+    protected final ContextService contextService;
+
+    protected final AclService aclService;
+
+    private final FieldFilterService fieldFilterService;
+
+    @GetMapping( produces = APPLICATION_JSON_VALUE )
+    @PreAuthorize( "hasRole('ALL') or hasRole('F_PERFORM_MAINTENANCE')" )
+    public @ResponseBody TrigramSummary getTrigramSummary( @RequestParam Map<String, String> rpParameters )
+    {
+        TrigramSummary trigramSummary = new TrigramSummary();
+
+        List<String> fields = new ArrayList<>( contextService.getParameterValues( "fields" ) );
+
+        if ( fields.isEmpty() )
+        {
+            fields.addAll( Preset.defaultPreset().getFields() );
+        }
+
+        Set<TrackedEntityAttribute> allIndexableAttributes = trackedEntityAttributeService
+            .getAllTrigramIndexableTrackedEntityAttributes();
+
+        Set<String> allIndexableAttributeUids = allIndexableAttributes.stream().map( TrackedEntityAttribute::getUid )
+            .collect( Collectors.toSet() );
+
+        List<Long> indexedAttributeIds = trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndexCreated();
```

Comment:
I know you did not add the method here. Just to mention that our vars and methods have very long names that can sometimes be shortened without loosing any meaningful information like `getAttributeIdsWithTrigramIndexCreated()` -> `getAttributeIdsWithTrigramIndex()`